### ISSUE TITLE
zbctl isn't OAuth spec compliant

### ISSUE THREAD

**Issue by dbarentine** [zeebe-io/zeebe#4026]
**Describe the bug**
Currently the zbctl client sends the OAuth token request payload as JSON. But the spec specifies this should be application/x-www-form-urlencoded. On spec compliant IDPs this causes the request to fail.

There is also a second problem where after successfully getting a token the client sends the header as:
`Authorization: bearer <jwt>`

But according to the spec this is case sensitive and should be:
`Authorization: Bearer <jwt>`

**To Reproduce**
Run the following against a spec compliant IDP. You can test it against a Keycloak server which is what we use for our STS.
`zbctl --address myzeebecluster.oneidentity.com:443 status --clientId my_client_id --clientSecret my_client_secret --authzUrl https://mysts/auth/realms/master/protocol/openid-connect/token`

**Expected behavior**
The client should follow the OAuth spec in both cases.

**Environment:**
 - OS: Ubuntu Linux and Windows
 - Zeebe CTL Version: 0.23.0-alpha.1

**Additional Information**
The code in question is in oauthCredentialsProvider.go and looks like Content-Type was set to application/json in #3781. Instead of using JSON the code should look something like the following to be spec compliant:

```golang
func (p *OAuthCredentialsProvider) buildOAuthRequest(ctx context.Context) (*http.Request, context.CancelFunc, error) {

    payload := url.Values{}
    payload.Set("grant_type", p.RequestPayload.GrantType)
    reader := strings.NewReader(payload.Encode())

    ctx, cancel := context.WithTimeout(ctx, p.timeout)

    req, err := http.NewRequestWithContext(ctx, "POST", p.AuthzServerURL, reader)
    if err != nil {
        cancel()
        return nil, nil, fmt.Errorf("failed while building request: %w", err)
    }

    req.SetBasicAuth(p.RequestPayload.ClientID, p.RequestPayload.ClientSecret)
    req.Header.Add("Content-Type", "application/x-www-form-urlencoded")

    return req, cancel, nil
}
```

The second issue is in ApplyCredentials in that same file:
```golang
func (p *OAuthCredentialsProvider) ApplyCredentials(ctx context.Context, headers map[string]string) error {
	credentials, err := p.getCredentials(ctx)
	if err != nil {
		return status.Errorf(codes.Canceled, err.Error())
	}

	headers["Authorization"] = fmt.Sprintf("%s %s", credentials.TokenType, credentials.AccessToken)
	return nil
}
```

Some quick research surfaced this: https://sgeb.io/posts/2015/05/fix-go-oauth2-case-sensitive-bearer-auth-headers/

So what I think is happening is our STS (Keycloak) is returning the token response with a lowercase "bearer" in it. Go is then using this directly. But our token validation happens in an Istio JWT/Authorization policy which follows the spec and is looking for "Bearer". If I change the code manually to "Bearer" it all works.

**Comment by menski** [comment]
@dbarentine thanks for raising this issue. Are you interested to provide a pull request to fix this issue?

@afirth could you verify if `application/x-www-form-urlencoded` is supported by our cloud proxy?

**Comment by afirth** [comment]
👍  thanks for the report!

- Our external IDP also specifies `application/x-www-form-urlencoded`. The parsing of json is working but unspecified behaviour
- This is a rehash of https://github.com/golang/oauth2/issues/113, fixed in https://github.com/golang/oauth2/commit/36ff901f7b5bce4fa4f23b4e0e02ab9a70ebf3ec - the RFCs conflict but the titlecase `Bearer` is a common requirement.

Replacing this implementation with calls to x/oauth2 would also fix #4035

**Comment by dbarentine** [comment]
@menski normally I would say yes but with my current work schedule it might be a while before I could put together a proper pull request with tests. I normally just dabble in Go so I'm sure it would take me longer than others that work in it more.

But when my schedule clears up, and if no one else has picked this up yet, I'll let you know and probably take a shot at it.
